api/v1alpha1/api.go

```diff
@@ -144,3 +144,71 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderType specifies the type of the LLMProviderPolicy.
+type LLMProviderType string
+
+const (
+	LLMProviderTypeAPIKey LLMProviderType = "APIKey"
+)
+
+// +kubebuilder:object:root=true
+
+// LLMProviderPolicy specifies the provider specific configuration.
+//
+// This is a provider specific-configuration, e.g.AWS Bedrock, Azure etc.
+type LLMProviderPolicy struct {
```

Comment:
i wonder at this poinst if `LLMProviderPolicy` is a write naming... maybe SecurityPolicy? Do we have anything other than security related stuff to this CRD? cc @wengyao04 @envoyproxy/ai-gateway-assignable

Reply:
At this moment, we are only setting auth. Not too familiar with what other provider specific config we would need. Alternatively to Security policy, we could revamp this to `LLMAuthPolicy` if nothing else is added? But it's very generic and not ai-gateway specific

Reply:
I also think SecurityPolicy is a better choice than ProviderPolicy because ProviderPolicy feels too generic atm. SecurityPolicy can handle not only authentication methods like auth, JWT, and API key, but it can also be extended to support future requirements like data masking or data protection algorithms etc. we may can keep LLM part as it is.

Reply:
I'll update this to be LLMSecurityPolicy if that works for everyone